torch/csrc/jit/script/compiler.cpp

```diff
@@ -552,25 +552,68 @@ struct to_ir {
     emitLoopCommon({}, {cond}, stmt.body(), stmt, {});
   }
 
+  size_t calcNumStarredUnpack(const List<Expr>& lhs, const Assign& stmt) {
+    size_t num_normal_assign = 0;
+    size_t num_starred = 0;
+    for (const auto& assignee : lhs) {
+      if (assignee.kind() == TK_VAR) {
+        num_normal_assign++;
+      } else if (assignee.kind() == '*') {
+        num_starred++;
+      } else {
+        throw ErrorReport(assignee)
+            << "lhs of assignment must be a variable or starred expression.";
+      }
+    }
+
+    if (num_starred > 1) {
+      throw ErrorReport(stmt)
+          << "Only one starred expression is allowed on the lhs.";
+    }
+
+    if (num_starred) {
+      return lhs.size() - num_normal_assign;
+    } else {
+      return 0;
+    }
+  }
+
   std::vector<Value*> emitAssignment(const Assign& stmt) {
     std::vector<Value*> outputs{stmt.lhs().size()};
+    size_t num_starred_unpack = calcNumStarredUnpack(stmt.lhs(), stmt);
```

Comment:
I don't think we will always be able to determine the size of the starred portion statically. It's definitely impossible if e.g. the rhs is a python call

Reply:
Design discussion: https://github.com/pytorch/pytorch/issues/6246